### PR TITLE
Warden failure app (v2)

### PR DESCRIPTION
**Pull request by dormi**
### Spec 

This is another attempt to https://github.com/dradis/dradis-ce/pull/432

When:
- we login with invalid credentials
- we make a POST/PATCH request and the session is expired 
We are redirected to the warden failure app. But since the request triggered in these cases check the csrf token, the failure app raises an error.

In this PR we try to set up a unique warden failure app, that does not check csrf token.
This failure app then just redirect to login for html requests, and for other requests (js/json) just returns 404 (not found)

### How to test
* Switch to `warden-failure-app-2` and confirm you now see a form when trying to login with bad credentials.
* In CE, create a node with subnodes (so when clicking the triangle to unfold the parent node a js request is generated). 
* Alter the app session timeout to be smaller. To do so edit this line: https://github.com/dradis/dradis-ce/blob/master/config/initializers/warden_99_timeout.rb#L30 so it looks like: 
  ` if last_request_at && last_request_at <= 10.seconds.ago`
and restart the server
* Login and visit a node so the attachments dropbox is visible. Wait > 10 seconds so the session expires.
 * When we try to upload an attachment, we should see an error like this:
captura de pantalla 2019-02-21 a las 20 41 39
* When we try to unfold a node in the sidebar we must see:
captura de pantalla 2019-02-21 a las 21 18 51
    
* Reload the page and assert we see the login form
* Login and visit an issue edit page, but don't save it. Wait > 10 seconds so the session expires and submit the form. We should be redirected to the login page.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -11,6 +11,7 @@
   # Sign in / sign out
   get '/login'  => 'sessions#new'
   get '/logout' => 'sessions#destroy'
+  match '/timeout' => 'sessions#timeout', via: :all
```
Comment (etdsoft) [inline config/routes.rb]: is this ever used?

File: config/routes.rb [hunk 2]
```diff
@@ -23,7 +24,7 @@
   resources :projects, only: [:show] do
     resources :activities, only: [] do
       collection do
-        get :poll, constraints: { format: /js/ }
+        get :poll, defaults: { format: :js }
```
Comment (etdsoft) [inline config/routes.rb]: this doesn't seem related? Why is it needed?

Comment (dormi) [reply config/routes.rb]: When I was checking with the browser console what was going on, I noticed that polling requests when the session expired tried to be redirected.
That is because rails was processing them as */* and in the failure app they were redirected as html format.
With this the poll requests is processed as js in the failure app the get a 404, because that is what we give to any html format.
Before, when a poll was received for and the session expired:
captura de pantalla 2019-02-21 a las 21 56 05
```
Processing by SessionsController#failure as */*
  Parameters: {"last_poll"=>"1550786143", "project_id"=>"1"}
Redirected to http://localhost:3000/login
Completed 302 Found in 1ms (ActiveRecord: 0.0ms)
```
And now it is:
captura de pantalla 2019-02-21 a las 21 57 24
```
Processing by SessionsController#failure as JS
  Parameters: {"last_poll"=>"1550786228", "project_id"=>"1"}
Completed 404 Not Found in 1ms (ActiveRecord: 0.0ms)
```

File: config/routes.rb (hunk 1 above)
Comment (dormi) [reply config/routes.rb]: no, you are right. Removed.

File: config/routes.rb (hunk 2 above)
Comment (dormi) [reply config/routes.rb]: ^ wait, scratch this please. I am seeing a different result when I try this on `master` branch now. The `poll` request gets the html login form now but I see no redirection.

Comment (dormi) [reply config/routes.rb]: Ok, I figured it out:

Now in `master` branch, when we get an expired `poll` request, it is processed by the current failure app `SessionController.new`, so it tries to render the login form with a `Session Expired` message:
captura de pantalla 2019-02-25 a las 10 39 30
(it is a js request and nobody sees that)

In this PR branch, we are introducing a new failure app: `SessionsController.failure`, that redirects expired html requests to login forms. That's why when we got an expired `poll` request, we were seeing a redirect:
captura de pantalla 2019-02-21 a las 21 56 05
The `poll` request format was not `js`, was `*/*`, so in the failure app it was handled as html.

With this change: `get :poll, defaults: { format: :js }` we are making the `poll` requests be processed as `js` requests:
captura de pantalla 2019-02-21 a las 21 57 24
Then when we get an expired `js` request we are just returning `head :unknown`.

Comment (etdsoft) [reply config/routes.rb]: I think the spirit of the original line was: only allow JS calls to this endpoint.

The problem was the client-side poller wasn't hitting `/poll.js` but `/poll`.

With this change, we're making any calls hitting `/poll` to be considered JS calls. Which works, however, is that what we want? I think that the other way to fix it (making the JS poller actually hit the JS endpoing) is preferable / clearer / more consistent with other JS calls (tree navigation for instance, console logger)?

In any case, we can move this to a separate PR, as the rest of the Warden-related code is ready to merge.

Comment (dormi) [reply config/routes.rb]: I understand what you say, and I think it was my other option to do it. I think it can be done like this: https://github.com/dradis/dradis-ce/pull/439/commits/3a4e524b4360b66b5f4030268e5186a2cc3212eb
